SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Utils/UserIdContext.cs

```diff
@@ -24,6 +24,13 @@ internal UserIdContext()
         /// </summary>
         public static int USER_ID_APP_CENTER_MAX_LENGTH = 256;
 
+        // The lock is static. Instance methods are not necessarily thread safe, but static methods are
+        private static readonly object UserIdContextLock = new object();
+
+        private string _userId = "";
+
+        public static event EventHandler<UserIdEventArgs> UserIdChangeReceived;
```

Comment:
Add some comment describe the event here. Something like 
"Event handler to subscribe to the user id update."